src/Workspaces/Core/Portable/Formatting/AutoFormattingOptions.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Options.Providers;
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    /// <summary>
+    /// Solution-wide formatting options.
+    /// </summary>
+    internal readonly record struct AutoFormattingOptions(
+        FormattingOptions.IndentStyle IndentStyle,
+        bool FormatOnReturn,
+        bool FormatOnTyping,
+        bool FormatOnSemicolon,
+        bool FormatOnCloseBrace)
```

Comment:
yikes.  we codified these c# concepts to the agnostic AutoFormattingOptions?  Is that how it was before as well?

Reply:
I'll follow up on moving these options to C#, if possible.

Reply:
https://github.com/dotnet/roslyn/issues/58917